### PR TITLE
#56 #38 Fix tests, test execution and enable test debugging

### LINKED ISSUE #38: Enable test debugging in VSCode
Enable test debugging in VSCode by adding necessary launch configs, mocha configs and dependencies (e.g. doublecheck with [`emfcloud-modelserver-theia`](https://github.com/eclipse-emfcloud/emfcloud-modelserver-theia/blob/master/.vscode/launch.json) repo).

### LINKED ISSUE #56: Fix tests and test execution
Currently, only very small subset of 4 tests is executed due to a misplaced `.only(..)` annotation.
Also, due to that inconsistency one test is failing due to the latest changes.

### PR DESCRIPTION
**Pull request by ndoschek**
- Fix test execution and failing test
  - Move tests to separate test directory
    - Adapt mocha config files
  - Extract helper methods to separate files
  - Remove misplaced `only` annotation in model-service integration test file

- Enable test debugging in VSCode
  - Add mocha config for debugging single files without timeout
  - Add launch config to debug currently opened file
  - Update build task, to build all packages

Contributed on behalf of STMicroelectronics

Resolves #56 
Resolves #38

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ module.exports = {
     },
     overrides: [
         {
-            files: ['*.spec.ts'],
+            files: ['*.spec.ts', 'test-*.ts'],
```
Comment (cdamus) [inline .eslintrc.js:12]: As tests are all now collected in a `test` directory, perhaps something like this?
```suggestion
            files: ['src/test/**'],
```

File: .vscode/launch.json [hunk 2]
```diff
@@ -25,6 +25,18 @@
       "console": "internalConsole",
       "internalConsoleOptions": "openOnSessionStart",
       "outputCapture": "std"
+    },
+    {
+      "type": "node",
+      "request": "launch",
+      "name": "Debug current test",
```
Comment (cdamus) [inline .vscode/launch.json:32]: What if the current editor is not a test file?

File: packages/modelserver-node/src/test/model-service-integration-test.spec.ts [hunk 3]
```diff
@@ -10,43 +10,40 @@
...
-import { awaitClosed, CoffeeMachine, isCoffeeMachine, ServerFixture } from '../server-integration-test.spec';
 import { TriggerProviderRegistry } from '../trigger-provider-registry';
 import { ValidationProviderRegistry } from '../validation-provider-registry';
+import {
+    _assumeThat,
+    awaitClosed,
+    findDiagnostic,
+    listenForFullUpdate,
+    registerCommand,
+    registerTrigger,
```
Comment (cdamus) [inline packages/modelserver-node/src/test/model-service-integration-test.spec.ts:35]: I'm not sure that these are useful in any other context. Did they need to be extracted to a separate file?

File: packages/modelserver-node/src/test/test-helpers.ts [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+        });
+    });
+}
+
+export function requireArray(owner: object, propertyName: string): unknown[] {
+    expect(owner[propertyName]).to.be.an('array').that.is.not.empty;
+    return owner[propertyName] as unknown[];
+}
+
+export function _assumeThat(this: Context, condition: boolean, reason: string): void {
```
Comment (cdamus) [inline packages/modelserver-node/src/test/test-helpers.ts:145]: If this is going to be an exported test API, it shouldn't start with an underscore.
```suggestion
export function assumeThat(this: Context, condition: boolean, reason: string): void {
```

**Review by cdamus — CHANGES_REQUESTED** [review]
Thanks, it's so nice to be able to debug the test code now. There's just a couple of nits that I would change.

File: .eslintrc.js (hunk 1 above)
Comment (ndoschek) [reply .eslintrc.js:12]: Thanks for catching that! Will update with upcoming commit.

File: .vscode/launch.json (hunk 2 above)
Comment (ndoschek) [reply .vscode/launch.json:32]: Then the debugger will quickly finish with `0 passing` if it cannot find test cases.
I am not aware of an easy way to restrict this, as the variable simply delivers the currently opened file (see also https://code.visualstudio.com/docs/editor/variables-reference#_predefined-variables).

File: packages/modelserver-node/src/test/model-service-integration-test.spec.ts (hunk 3 above)
Comment (ndoschek) [reply packages/modelserver-node/src/test/model-service-integration-test.spec.ts:35]: Agreed, moved them back again.

File: packages/modelserver-node/src/test/test-helpers.ts (hunk 4 above)
Comment (ndoschek) [reply packages/modelserver-node/src/test/test-helpers.ts:145]: Good catch, agreed!

**Review by cdamus — APPROVED** [review]
Thanks @ndoschek ! LGTM.

**Pull request merged**